### ISSUE TITLE
[TRT] [E] plugin/instanceNormalizationPlugin/instanceNormalizationPlugin.cu (335) - Cudnn Error in enqueue: 8 (CUDNN_STATUS_EXECUTION_FAILED) failure of TensorRT 8.6.1  when running FastReid model on RTX4090

### ISSUE THREAD

**Issue by Merealtea** [NVIDIA/TensorRT#3262]
## Description

I try to use the torch2trt to turn [fast-reid](https://github.com/JDAI-CV/fast-reid/blob/master/configs/MSMT17/sbs_R50-ibn.yml) model to trt model, and the code is :

fast_reid = FastReid().to(device).eval()
x = torch.ones((64, 3, 256, 256)).to(device)
fast_reid.model.net = torch2trt(fast_reid.model.net, [x], int8_mode=False, fp16_mode=False, use_onnx=True, max_batch_size = 128) 

And when beginning the inference, the error appeared :

> /home/cyber-fx/miniconda3/envs/cityvehicletracking/lib/python3.8/site-packages/torch/onnx/symbolic_helper.py:1457: UserWarning: ONNX export mode is set to TrainingMode.EVAL, but operator 'instance_norm' is set to train=True. Exporting with train=True.
>   warnings.warn(
> /home/cyber-fx/miniconda3/envs/cityvehicletracking/lib/python3.8/site-packages/torch/onnx/_internal/jit_utils.py:258: UserWarning: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function. (Triggered internally at /opt/conda/conda-bld/pytorch_1666642975312/work/torch/csrc/jit/passes/onnx/shape_type_inference.cpp:1884.)
>   _C._jit_pass_onnx_node_shape_type_inference(node, params_dict, opset_version)
> /home/cyber-fx/miniconda3/envs/cityvehicletracking/lib/python3.8/site-packages/torch/onnx/utils.py:687: UserWarning: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function. (Triggered internally at /opt/conda/conda-bld/pytorch_1666642975312/work/torch/csrc/jit/passes/onnx/shape_type_inference.cpp:1884.)
>   _C._jit_pass_onnx_graph_shape_type_inference(
> /home/cyber-fx/miniconda3/envs/cityvehicletracking/lib/python3.8/site-packages/torch/onnx/utils.py:1178: UserWarning: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function. (Triggered internally at /opt/conda/conda-bld/pytorch_1666642975312/work/torch/csrc/jit/passes/onnx/shape_type_inference.cpp:1884.)
>   _C._jit_pass_onnx_graph_shape_type_inference(
> [W] colored module is not installed, will not use colors when logging. To enable colors, please install the colored module: python3 -m pip install colored
> [W] Inference failed. You may want to try enabling partitioning to see better results. Note: Error was:
> [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Failed to load model with error: /onnxruntime_src/onnxruntime/core/graph/model.cc:145 onnxruntime::Model::Model(onnx::ModelProto&&, const PathString&, const IOnnxRuntimeOpSchemaRegistryList*, const onnxruntime::logging::Logger&, const onnxruntime::ModelOptions&) Unsupported model IR version: 9, max supported IR version: 8
> 2339it [00:01, 999.36it/s][08/23/2023-20:42:05] [TRT] [E] plugin/instanceNormalizationPlugin/instanceNormalizationPlugin.cu (335) - Cudnn Error in enqueue: 8 (CUDNN_STATUS_EXECUTION_FAILED)

## Environment

System : Uubuntu 20.04

cuda version : 12.1

cudnn version ： I don`t know what my cudnn version is, but the installation package I used is cudnn-linux-x86_64-8.9.3.28_cuda12-archive.tar.xz, and the pytorch shows that the cudnn version is 8500

GPU : RTX 4090 24G

tensortrt version : 8.6.1.6

pytorch version: 1.13.1+cu117

torch2trt version : 0.4.0

python version : 3.8.10

## Relevant Files

The FastReid Model configure link is https://github.com/JDAI-CV/fast-reid/blob/master/configs/MSMT17/sbs_R50-ibn.yml

By the way, I only use this model for offline inference, so I turn it into the eval mode for inference, And I also want to use this model for dynamic shape inference due to the batchsize is always changing. Thx in advance!

**Comment by Merealtea** [comment]
In addition, my onnx version is 1.14.0 , onnxruntime version is 1.15.1

**Comment by zerollzeng** [comment]
> Cudnn Error in enqueue: 8 (CUDNN_STATUS_EXECUTION_FAILED)

Looks like a cuDNN installation issue to me, could you please try our docker as test env? See https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch

**Comment by Merealtea** [comment]
Thank you for your reply! And yes, the container of version pytorch:23.01-py3 did work for me. Can I build a image based on this image, or is there any solution for cudnn installation? I followed the  installation guide here [https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#installlinux-tar](url) in section 1.3.1, but it seems my cudnn version changed nothing. So what is the proper way to install the cudnn? I notice that the cudnn version in the pytorch:23.01-py3 container is 8700

**Comment by Merealtea** [comment]
Thank you @zerollzeng  for your reply! And yes, the container of version pytorch:23.01-py3 did work for me. Can I build a new image based on this image, or is there any solution for cudnn installation? I followed the  installation guide here [https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#installlinux-tar](url) in section 1.3.1, but it seems my cudnn version changed nothing. So what is the proper way to install the cudnn? I notice that the cudnn version in the pytorch:23.01-py3 container is 8700

**Comment by zerollzeng** [comment]
> but it seems my cudnn version changed nothing.

Did you uninstall the old one before you install the new package?

> Can I build a new image based on this image

Yes

**Comment by Merealtea** [comment]
I will try both of the advice. thank you！

**Comment by Merealtea** [comment]
I tried the both of the approach. Building a new image based on officical tensorRT image did work for me, but installing a new cudnn failed. I removed the old cudnn and installed the new cudnn by using debian and tar.xz file, There was no version change that pytorch shows and the error still appeared. Maybe I did`t remove the old cudnn clearly? Anyway, I could use the image as the solution and also I will figure out how to install cudnn correctly. Case close! Thank you so much! ]
